### PR TITLE
Move MVC's fallback `IHttpSendFileFeature` and `SendFileMiddleware` to this repo

### PR DESCRIPTION
**Pull request by dougbu**
- related to aspnet/Mvc#3612
- create sample

### CODE REVIEW

File: samples/FallbackSendFile/SendFileMiddleware.cs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Http.Features;
+using Microsoft.Extensions.PlatformAbstractions;
+
+namespace FallbackSendFile
+{
+    /// <summary>
+    /// Middleware which adds an <see cref="IHttpSendFileFeature"/> implementation if currently missing.
+    /// </summary>
+    public class SendFileMiddleware
```
Comment (dougbu) [inline samples/FallbackSendFile/SendFileMiddleware.cs]: Do not plan to add tests of this sample. That would require odd test dependencies e.g. on `TestHost`. And Kestrel, TestHost and WebListener all support `IHttpSendFileFeature` without any help.

Comment (davidfowl) [inline samples/FallbackSendFile/SendFileMiddleware.cs]: Why is this here? We removed this middleware completely and made it an extension method in Microsoft.AspNet.Http.Extensions.

Comment (dougbu) [inline samples/FallbackSendFile/SendFileMiddleware.cs]: This is coming from MVC/test/WebSites/FilesWebSite. The request in aspnet/Mvc#3612 was to move it elsewhere.

Could just delete this middleware and the feature implementation from MVC repo if you prefer. It's certainly unneeded in the MVC repo.

Comment (davidfowl) [inline samples/FallbackSendFile/SendFileMiddleware.cs]: Delete it

**Comment by dougbu** [comment]
Not going forward with this PR. As @davidfowl requested, won't move `SendFileMiddleware` anywhere.
